### PR TITLE
fix: only one warning about skip resolution

### PR DESCRIPTION
**Pull request by jayahariv**
* only one log per conflict skipping.

### CODE REVIEW

File: Objective-C/CBLReplicator.mm [hunk 1]
```diff
@@ -597,7 +597,7 @@ - (void) _resolveConflict: (CBLReplicatedDocument*)doc {
                                withConflictResolver: _config.conflictResolver
                                               error: &error])
         [doc resetError];
-    else {
+    else if (error) {
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm]: This means that if the error is nil, we have never reseted the error. I think we probably don't need the `resetError` method if we always update the error at the end of the `_resolveConflict:` method.

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: then i feel like we should be returning success, else seems like we will end up with 


Screen Shot 2019-06-19 at 10 14 43 AM

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: updated the change: https://github.com/couchbase/couchbase-lite-ios/pull/2515/commits/154b2e7240103a90d68086556f46fc3c282a705b

File: Objective-C/CBLReplicator.mm [hunk 2]
```diff
@@ -593,14 +593,16 @@ - (void) resolveConflict: (CBLReplicatedDocument*)doc {
 - (void) _resolveConflict: (CBLReplicatedDocument*)doc {
     CBLLogInfo(Sync, @"%@: Resolve conflicting version of '%@'", self, doc.id);
     NSError* error;
-    if ([_config.database resolveConflictInDocument: doc.id
-                               withConflictResolver: _config.conflictResolver
-                                              error: &error])
+    BOOL success = [_config.database resolveConflictInDocument: doc.id
+                                          withConflictResolver: _config.conflictResolver
+                                                         error: &error];
+    if (success && !error) {
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm]: Is there a case of success and having an error?

Comment (pasin) [reply Objective-C/CBLReplicator.mm]: Can we do something this?

```
- (void) _resolveConflict: (CBLReplicatedDocument*)doc {
    CBLLogInfo(Sync, @"%@: Resolve conflicting version of '%@'", self, doc.id);
    NSError* error = nil;
    BOOL success = [_config.database resolveConflictInDocument: doc.id
                                          withConflictResolver: _config.conflictResolver
                                                         error: &error];
    if (!success)
        CBLWarn(Sync, @"%@: Conflict resolution of '%@' failed: %@", self, doc.id, error);
    
    [doc updateError: error];
    [self logErrorOnDocument: doc pushing: NO];
    [self postDocumentReplications: @[doc] pushing: NO];
}
```

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: i think, if we return true from the database, this is enough.

Comment (jayahariv) [reply Objective-C/CBLReplicator.mm]: > Is there a case of success and having an error?
there is none i can think of. i think, only checking success is fine

Comment (pasin) [reply Objective-C/CBLReplicator.mm]: Then in updateError, if error is nil, you can internally resetError so don't need two methods, resetError and updateError.

**Pull request merged**